Elmanager/ElmanagerSettings.cs

```diff
@@ -227,9 +228,51 @@ public class ReplayManagerSettings : ManagerSettings
             public string Nickname = "Nick";
             public bool NitroReplays;
             public string Pattern = "LNT";
+            public string LevPattern = "";
+            
+            private int? _intExtSelect = 2;
+            private int? _singleMultiSelect = 2;
+            private int? _finishedSelect = 2;
+            private int? _fastestSlowestSelect = 2;
+            private int? _elmaAcrossSelect = 2;

```

Comment:
Yeah, as you noted, these nullables don't look so good. I guess part of the blame is the `BinaryFormatter` because the class constructor does not get run upon deserialization, so you can't control the default values properly. This would not be a problem with XML deserialization ([source](https://stackoverflow.com/questions/3492263/using-xml-decorations-to-specify-default-values-during-de-serialization)).

Using `BinaryFormatter` has been a mistake anyway because it's also [insecure](https://docs.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-security-guide).

I would accept the following kind of changes:

* Elmanager would use XML deserialization by default, which means:
  * Instead of looking for dat files, it first tries to find `Elmanager*.xml` files.
  * If found, it uses XML deserialization for that.
  * Otherwise, it falls back to searching for `Elmanager*.dat` files and uses binary deserialization.
  * When saving settings, it always serializes to XML, not binary.
* The `*Select` attributes would not be nullable.
  * Upon XML deserialization, the constructor is called so the default values for `Select`s can be assigned in the constructor (or directly in the field declaration) normally.
  * Upon binary deserialization, just assign the default values for `Select`s in `OnDeserialized` unconditionally. This should be fine because the next time the settings will be loaded from XML, so the binary `OnDeserialized` won't be called anymore.

And sometime in the future, the binary deserialization can be completely removed.

Instead of XML, JSON is okay too if it's easier to implement. For now, Elmanager needs to work on .NET Framework too (because the .NET Core WinForms designer doesn't work well enough yet), so JSON might be problematic.

Reply:
I started working on JSON serialization. I dropped .NET Framework support a while ago, so no need for XML.